packages/eas-cli/src/graphql/queries/ChannelQuery.ts

```diff
@@ -59,7 +60,7 @@ export const ChannelQuery = {
 
     const { updateChannelByName } = response.app.byId;
```

Comment:
slight preference for making this return nullable so that no special error type or type introspection is needed:

```
async viewUpdateChannelAsync(...): Promise<UpdateChannelByNameObject | null> {
...
}

async viewUpdateChannelEnforcingAsync(...): Promise<UpdateChannelByNameObject> {
     const data = await viewUpdateChannelAsync(...);
     if (!data) {
        throw new Error(`Could not find channel with the name ${channelName}`);
     }
     return data;
}
```

Reply:
This special error type is used elsewhere. We have this logic:

```ts
try {
// ...
} catch (error) {
  if (error instanceof ChannelNotFoundError) {
    // create the channel
  }

  throw error; // display any other error
}
```

Are you suggesting we do a different pattern? (Also, this is how branches work in this code base, so I used that as my pattern. If we change here, we ought to change there too)